client/state/current-user/gravatar-status/actions.js

```diff
@@ -34,20 +41,40 @@ export function uploadGravatar( file, bearerToken, email ) {
 					dispatch( {
 						type: GRAVATAR_UPLOAD_REQUEST_SUCCESS
 					} );
+					dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_success' ) );
 				} );
 				fileReader.readAsDataURL( file );
 			} )
 			.catch( () => {
 				dispatch( {
 					type: GRAVATAR_UPLOAD_REQUEST_FAILURE
 				} );
+				dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_failure' ) );
+				dispatch( bumpStat( 'calypso_gravatar_update_error', 'unsuccessful_http_response' ) );
 			} );
 	};
 }
 
-export function receiveGravatarImageFailed( errorMessage ) {
-	return {
-		type: GRAVATAR_RECEIVE_IMAGE_FAILURE,
-		errorMessage
+export function receiveGravatarImageFailed( { errorMessage, statName } ) {
+	return dispatch => {
+		dispatch( recordTracksEvent( 'calypso_edit_gravatar_file_recieve_failure' ) );
+		dispatch( bumpStat( 'calypso_gravatar_update_error', statName ) );
+		dispatch( {
+			type: GRAVATAR_RECEIVE_IMAGE_FAILURE,
+			errorMessage,
+		} );
+	};
+}
+
+export function clickButton( { isVerified } ) {
+	return dispatch => {
+		dispatch( recordTracksEvent( 'calypso_edit_gravatar_click', { userVerified: isVerified } ) );
+		dispatch( recordGoogleEvent( 'Me', 'Clicked on Edit Gravatar Button in Profile' ) );
```

Comment:
Didn't end up using `ownProps` since the `user` prop is coming from `mapStateToProps` and is not available in `ownProps`, so we'd to add the `mergeProps` argument. I opted for a parameter on the click button fn as before.